src/react-native-animated-charts/src/charts/linear/ChartPathProvider.tsx

```diff
@@ -0,0 +1,206 @@
+import { scaleLinear } from 'd3-scale';
+import * as shape from 'd3-shape';
+import React, {
+  useCallback,
+  useMemo,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { Dimensions } from 'react-native';
+import { useSharedValue } from 'react-native-reanimated';
+import * as redash from 'react-native-redash';
+import {
+  CallbackType,
+  ChartContext,
+  CurveType,
+  DataType,
+  PathData,
+  Point,
+  PathScales,
+} from '../../helpers/ChartContext';
+import { findYExtremes } from '../../helpers/extremesHelpers';
+
+export const { width: WIDTH } = Dimensions.get('window');
+const HEIGHT = 146.5;
+
+interface ChartPathProviderProps {
+  data: DataType;
+  width?: number;
+  height?: number;
+  yRange?: [number, number];
+}
+
+function getCurveType(curveType: CurveType) {
+  switch (curveType) {
+    case CurveType.basis:
+      return shape.curveBasis;
+    case CurveType.bump:
+      return shape.curveBumpX;
+    case CurveType.linear:
+      return shape.curveLinear;
+    case CurveType.monotone:
+      return shape.curveMonotoneX;
+    case CurveType.natural:
+      return shape.curveNatural;
+    case CurveType.step:
+      return shape.curveStep;
+
+    default:
+      return shape.curveBasis;
+  }
+}
+
+export const ChartPathProvider = React.memo<ChartPathProviderProps>(
+  ({ children, data, width = WIDTH, height = HEIGHT, yRange }) => {
+    const progress = useSharedValue(1);
+    const dotScale = useSharedValue(0);
+    const isActive = useSharedValue(false);
+    const originalX = useSharedValue('');
+    const originalY = useSharedValue('');
+    const pathOpacity = useSharedValue(1);
+    const state = useSharedValue(0);
+    const positionX = useSharedValue(0);
+    const positionY = useSharedValue(0);
+
+    const getScales = useCallback(
+      ({ data, width, height, yRange }: CallbackType): PathScales => {
+        const x = data.points.map(item => item.x);
+        const y = data.points.map(item => item.y);
+
+        const smallestX = Math.min(...x);
+        const smallestY = Math.min(...y);
+        const greatestX = Math.max(...x);
+        const greatestY = Math.max(...y);
+
+        const scaleX = scaleLinear()
+          .domain([smallestX, greatestX])
+          .range([0, width]);
+
+        const scaleY = scaleLinear()
+          .domain(yRange ?? [smallestY, greatestY])
+          .range([height, 0]);
+
+        return {
+          scaleY,
+          scaleX,
+        };
+      },
+      []
+    );
+
+    const createPath = useCallback(
```

Comment:
and this? this doesn't need to be inside the component, right?

Reply:
Makes sense